tests/test_write_nwb.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import datetime
+import pynwb
+import h5py
+
+from ipfx.x_to_nwb.NWBConverter import NWBConverter
+from .helpers_for_tests import diff_h5, validate_nwb
+from ipfx.bin.run_nwb1_to_nwb2_conversion import make_nwb2_file_name
+from hdmf import Container
+import numpy as np
+import ipfx.nwb_reader as nwb_reader
+
+from ipfx.bin.run_feature_extraction import embed_spike_times
+
+
+
+
```

Comment:
I think the functional implementation is clearer.
I would recommend making use of pytest fixtures and parametrization. Since `test_embed_spike_times_into_nwb2` and `test_embed_spike_times_into_nwb1` are exactly the same outside of arguments and filenames, you can just have one function with the inputs defined using the `pytest.mark.parametrize`  decorator.

Reply:
I agree, the function implementation looks clearer. I am just not sure how to implement  this test with a single function since nwb1 and nwb2 files are created by `make_skeleton_nwb1_file` and `make_skeleton_nwb2_file`, respectively. Can I pass different fixtures to the same test function?

Reply:
You could write something like:
```
@pytest.mark.parametrize("make_skeleton_file", [make_skeleton_nwb1_file, make_skeleton_nwb2_file])
def test_embed_spike_times(make_skeleton_file):
    ...
    make_skeleton_file(input_nwb_file_name)
    ...
```

Also, the `make_skeleton_nwb\d_file` functions are not actually pytest fixtures.

Reply:
done